atoma-auth/src/auth.rs

```diff
@@ -355,6 +369,69 @@ impl Auth {
             })?;
         Ok(result_receiver.await??)
     }
+
+    /// Stores the wallet address for the user. The user needs to send a signed message to prove ownership of the wallet.
+    /// The wallet address is stored in the signature.
+    ///
+    /// # Arguments
+    /// * `jwt` - The access token to be used to store the wallet address
+    /// * `signature` - The signature of the message
+    ///
+    /// # Returns
+    ///
+    /// * `Result<()>` - If the wallet address was stored
+    #[instrument(level = "info", skip(self))]
+    pub async fn update_sui_address(&self, jwt: &str, signature: &str) -> Result<()> {
+        let claims = self.validate_token(jwt, false)?;
+        let signature = Signature::from_str(signature).map_err(|e| {
+            error!("Failed to parse signature: {}", e);
+            anyhow::anyhow!("Failed to parse signature {e}")
+        })?;
+        let signature_bytes = signature.signature_bytes();
+        let public_key_bytes = signature.public_key_bytes();
+        let signature_scheme = signature.scheme();
+        let intent_msg = IntentMessage::new(
+            Intent::personal_message(),
+            PersonalMessage {
+                message: "Sign this message to prove you are the owner of this wallet"
```

Comment:
Hardcoded messages signign is generally not secure

Reply:
added user_id. It has to match the access_token user_id.